etna/records/models.py

```diff
@@ -1,78 +1,333 @@
-    data as though the data was fetched from the CMS.
-
-    see: views.record_detail_view
-    """
-
-    iaid: str
-    title: str
-    reference_number: str
-
-    legal_status: str = ""
-    created_by: str = ""
-    description: str = ""
-    origination_date: str = ""
-    closure_status: str = ""
-    availability_delivery_condition: str = ""
-    arrangement: str = ""
-    held_by: str = ""
-    is_digitised: bool = False
-    parent: dict = field(default_factory=dict)
-    hierarchy: dict = field(default_factory=dict)
-    media_reference_id: str = ""
-    availability_delivery_surrogates: dict = field(default_factory=dict)
-    topics: dict = field(default_factory=dict)
-    next_record: dict = field(default_factory=dict)
-    previous_record: dict = field(default_factory=dict)
-    related_records: dict = field(default_factory=dict)
-    related_articles: dict = field(default_factory=dict)
-    catalogue_source: str = ""
-    repo_summary_title: str = ""
-    repo_archon_value: str = ""
-    level_code: str = ""
-    level: str = ""
-    template_reference_number: str = ""
-    template_summary_title: str = ""
-    hierarchy_level3_reference_number: str = ""
-    hierarchy_level3_summary_title: str = ""
-    related_materials: list = field(default_factory=list)
-
-    _debug_kong_result: dict = field(default_factory=dict)
+    """A 'lazy' data-interaction layer for record data retrieved from the Kong API"""
+
+    def __init__(self, raw_data: Dict[str, Any]):
+        """
+        This method recieves the raw JSON data dict recieved from
+        Kong and makes it available to the instance as `self._raw`.
+        """
+        self.score = raw_data.get("_score")
+        self._raw = raw_data.get("_source") or raw_data
+
+    @classmethod
+    def from_api_response(cls, response: dict) -> Record:
+        return cls(response)
 
     def __str__(self):
         return f"{self.title} ({self.iaid})"
 
-    def get(self, key: str, default: Optional[Any] = None):
-        return getattr(self, key, default)
+    def get(self, key: str, default: Optional[Any] = NOT_PROVIDED):
+        """
+        Attempts to extract `key` from `self._raw` and return the value.
 
-    @classmethod
-    def from_api_response(cls, response: dict) -> Record:
-        return cls(**transform_record_result(response))
+        Raises `ciim.utils.ValueExtractionError` if the value cannot be extracted.
+        """
+        if "." in key:
+            return extract(self._raw, key, default)
+        try:
+            return self._raw[key]
+        except KeyError as e:
+            if default is NOT_PROVIDED:
+                raise ValueExtractionError(str(e))
+            return default
+
+    @cached_property
+    def template(self) -> Dict[str, Any]:
+        return self.get("@template.details", default=self.get("@template.results", {}))
+
+    @cached_property
+    def highlights(self) -> Dict[str, Any]:
+        return self.get("highlight", default={})
```

Comment:
Where is this property used? What API query will return data for highlight?

Reply:
This is supplied when you do a text search... it brings back values with `<mark></mark>` applied to indicate where the terms matched for that result